setup/building-stratum-deps.md

```diff
@@ -0,0 +1,183 @@
+an RPM file, a Debian file, or a tarball.
+
+If this is case, use the appropriate command to unpack the source files.
+
+## 3 Install Location
+
+You will need to decide where on your system you would like to install the
+dependencies. This location (the "install prefix") must be specified when you
+configure the build.
+
+It is recommended that you _not_ install the dependency libraries in `/usr` or
+`/usr/local`. It is easier to upgrade or do a clean build if packages that are
+updated on different cadences are kept separate from one another.
+
+The `CMAKE_INSTALL_PREFIX` option is used to specify the directory in which
+the dependencies should be installed.
+
+If you plan to install the dependency libraries in a system directory, you will
+need to log in as `root` or run from an account that has `sudo` privilege.
+
+## 4 Downloaded Source vs. Source Package
+
+By default, the cmake build script downloads the source code for each module
+and then builds and installs it.
+
+If you are working from a source distribution, or if the source files have
+were downloaded by a previous build, you can override the download stage by
+specifying:
+
+```cmake
+  -DDOWNLOAD=no -DFORCE_PATCH=yes
+```  
+
+when you configure the cmake build.
+
+## 5 Build Options
+
+The CMake build script supports the following configuration options.
+
+| Option | Type | Description |
+| ------ | ---- | ----------- |
+| `CMAKE_INSTALL_PREFIX` | Path | Directory in which dependencies should be installed. |
+| `CXX_STANDARD` | Number | C++ standard (11, 14, 17, etc.) the compiler should apply. (Default: not specified) |
+| `DOWNLOAD` | Boolean | Whether to download the source repositories. (Default: TRUE)
+| `FORCE_PATCH` | Boolean | Whether to specify the force (`-f`) option when patching a downloaded repository. (Default: FALSE) |
+| `ON_DEMAND` | Boolean | Whether to build only the specified target(s). If this option is FALSE, all targets will be built. (Default: FALSE) |
+| `USE_LDCONFIG` | Boolean | Whether to use `ldconfig` to update the loader cache after installing a module. Only valid if `USE_SUDO` is enabled. (Default: FALSE) |
+| `USE_SUDO` | Boolean | Whether to use `sudo` to install each module. (Default: FALSE) |
+
+Boolean values are (`TRUE`, `YES`, `ON`) and (`FALSE`, `NO`, `OFF`).
+They may be upper or lower case.
+
+## 6 Examples
+
+### 6.1 Non-root build
+
+To build and install as a nonprivileged user:
+
+```bash
+cd setup
+cmake -B build -DCMAKE_INSTALL_PREFIX=./install
+cmake --build build -j8
+```
+
+The source files will be downloaded, built, and installed in the `setup/install`
+directory. The targets will be built in parallel, using eight threads.
+
+CMake will generate all of its temporary files in the `setup/build` directory.
+This is called an "out-of-source" build.
+
+### 6.2 Non-root build to system directory
+
+To build as a nonprivileged user and use `sudo` to install the libraries to
```

Comment:
`non-privileged`? (with hyphen)

Reply:
Interesting. A web search turns up `nonprivileged`, `non-privileged`, and `unprivileged`.

1. Looking at the first two side-by-side, I find `non-privileged` somewhat easier to read than `nonprivileged`, which is reason enough to prefer it.
2. One of the first hits I got brought me to `unprivileged`, suggesting that the unhyphenated form is less common or less correct.
3. `unprivileged` might be more correct, but has a pejorative connotation that I'd like to avoid.

`non-privileged` it is.